ldap/servers/slapd/conntable.c

```diff
@@ -48,69 +48,61 @@
  * under the connection table lock. This should move the allocation algorithm from O(n) worst case
  * to O(1) worst case as we always recieve and empty slot *or* ct full. It also reduces lock/atomic
  * contention on the CPU to improve things.
+ * Lastly a change was done: Instead of having a sliding windows that tend to get a never used
+ * slot for each new connection, it nows reuse last freed one. That has several benefits:
+ *  - Fix a bug because the last free list slaots may not be alloced.
```

Comment:
Why were the last slots not allocated?

Reply:
Mostly because I saw it was the case when gdb triggered the watch point ! -;) 
More seriously: 
  I initially thought it was because of rounding issue (i.e: table_size != ct->list_size*ct->numlist)
but the way initialization is done, table_size >= ct->size*ct->numlist so there should not be any empty slot.
So I think that the reason is that we skip first slot for each listener table when filling the free list  
(I suspect the slot it is reserved (maybe for the signal pipe ? )